pkg/operator/ceph/object/rgw.go

```diff
@@ -137,11 +144,9 @@ func (c *clusterConfig) startRGWPods() error {
 			Name:       rgwConfig.ResourceName,
 		}
 
-		// Generate the keyring after starting the replication controller so that the keyring may use
-		// the controller as its owner reference; the keyring is deleted with the controller
-		err = c.generateKeyring(resourceControllerOwnerRef)
+		err = c.associateKeyring(keyring, resourceControllerOwnerRef)
 		if err != nil {
-			return fmt.Errorf("failed to create rgw keyring. %+v", err)
+			return fmt.Errorf("failed to associate keyring with rgw deployment %q: %+v", createdDeployment.Name, err)
```

Comment:
Does this have to be a fail-able error? I think it might be sufficient to use `logger.Warningf` or `logger.Errorf`. It's definitely non-idea if this isn't associated, but it won't stop Rook/Ceph from functioning in the long term, and this operation would be retried on future orchestrations. There is increased risk for hanging secrets with no parent, however. @travisn thoughts?

Reply:
Agreed, if the association fails, functionality isn't blocked. Sounds good to just log the error.

Reply:
Sure; makes sense. I considered this, and was mostly concerned that allowing variation among the ownership structure of a cluster could cause issues in tooling created by end users which might depend on a specific hierarchy: I imagined that in this hopefully rare case it might be best to abort. This is another issue surrounding where our contract ends and mutable implementation details begin; good to know that even within versions of Rook, we don't always consider structural changes in resource hierarchy a stark failure case. Thanks!